mail_tracking/models/ir_mail_server.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antonio Espinosa - <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import re
+from openerp import models, api, tools
+
+
+class IrMailServer(models.Model):
+    _inherit = "ir.mail_server"
+
+    def _tracking_headers_add(self, tracking_email_id, headers):
+        """Allow other addons to add its own tracking SMTP headers"""
+        headers = headers or {}
+        headers['X-Odoo-Tracking-ID'] = tracking_email_id
```

Comment:
@lasley This header is only used just before email is sent, in order to recover the tracking object created when the email was created. As @pedrobaeza said, this trick is used because there is no way to do using context.

This header is not read when the email is fetched by the same Odoo using fetchmail, so there will not be cross-talk. 

BTW, it would be great if you test it and share the results.